### PR TITLE
HTTP post precompiles

### PR DESCRIPTION
**Pull request by kziemianek**
Adds precompiles for making HTTP Post requests.

### CODE REVIEW

File: tee-worker/litentry/core/assertion-build/src/dynamic/contracts/tests/PostBool.sol [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
+
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.8;
+
+import {DynamicAssertion, Identity, HttpHeader} from "./DynamicAssertion.sol";
```
Comment (Kailai-Wang) [inline tee-worker/litentry/core/assertion-build/src/dynamic/contracts/tests/PostBool.sol]: I have been wondering why not use `../DynamicAssertion.sol`

Or having a symlink if it's meant to be under the same folder

File: tee-worker/litentry/core/assertion-build/src/dynamic/contracts/tests/PostBool.sol [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    function callPostBoolTwiceAndReturnSecondResult(
+        string memory firstUrl,
+        string memory firstJsonPointer,
+        string memory firstPayload,
+        string memory secondUrl,
+        string memory secondJsonPointer,
+        string memory secondPayload
+    ) public returns (bool, bool) {
+        HttpHeader[] memory headers = new HttpHeader[](0);
+        PostI64(firstUrl, firstJsonPointer, firstPayload, headers);
```
Comment (Kailai-Wang) [inline tee-worker/litentry/core/assertion-build/src/dynamic/contracts/tests/PostBool.sol]: Should it be `PostBool` here?

**Review by Kailai-Wang — COMMENTED** [review]
The systematic unit-tests are particularly appreciated

File: tee-worker/litentry/core/assertion-build/src/dynamic/contracts/tests/PostBool.sol (hunk 1 above)
Comment (kziemianek) [reply tee-worker/litentry/core/assertion-build/src/dynamic/contracts/tests/PostBool.sol]: fixed

File: tee-worker/litentry/core/assertion-build/src/dynamic/contracts/tests/PostBool.sol (hunk 2 above)
Comment (kziemianek) [reply tee-worker/litentry/core/assertion-build/src/dynamic/contracts/tests/PostBool.sol]: fixed

**Comment by kziemianek** [comment]
Let's merge #2785 first.

**Comment by wli-pro** [comment]
LGTM

**Review by wli-pro — APPROVED** [review]

**Pull request merged**